src/stdlib/RandomMeasures.jl

```diff
@@ -128,41 +128,31 @@ function distribution(d::SizeBiasedSamplingProcess{<:DirichletProcess})
     return LocationScale(zero(α), d.surplus, Beta(one(α), α))
 end
 
-function _logpdf_table(d::DirichletProcess, m::AbstractVector{Int})
+function _logpdf_table(d::DirichletProcess{T}, m::AbstractVector{Int}) where {T<:AbstractFloat}
```

Comment:
The static type parameter is not needed (since `T = typeof(d.α)`), but in any case I guess it should be `T<:Real` since `d.α isa Real` but not necessarily an `AbstractFloat`.

Reply:
I think we should make `T<:AbstractFloat`. `Real` doesn't actually make sense to me. I'll change the type signature.

Reply:
IMO one should keep `Real` - using `AbstractFloat` would remove the possibility to use `Integer`s, dual numbers from ForwardDiff, and tracked reals from Tracker as concentration parameter.